asdfghjkl/precondition.py

```diff
@@ -66,15 +66,15 @@ def update_curvature(self, inputs=None, targets=None, data_loader=None):
                                        inputs=inputs,
                                        targets=targets,
                                        data_loader=data_loader,
-                                       fisher_types=self.fisher_type,
+                                       fisher_type=self.fisher_type,
                                        fisher_shapes=self.fisher_shape,
                                        n_mc_samples=self.n_mc_samples)
         self.fisher_manager = rst
 
-    def move_curvature(self, postfix, scale=1., to_pre_inv=False):
-        self.accumulate_curvature(postfix, scale, to_pre_inv, replace=True)
+    def move_curvature(self, postfix, scale=1., ema_decay=None, to_pre_inv=False):
+        self.accumulate_curvature(postfix, scale, ema_decay, to_pre_inv, replace=True)
 
-    def accumulate_curvature(self, postfix='acc', scale=1., to_pre_inv=False, replace=False):
+    def accumulate_curvature(self, postfix='acc', scale=1., ema_decay=None, to_pre_inv=False, replace=False):
         if to_pre_inv:
```

Comment:
thanks for adding a feature of EMA! this is useful.

can you add a logic like:
> if ema_decay is None:
>    ema_decay = self.ema_decay

(same as the `damping` in `update_inv`)
using `self.ema_decay` (default=None) is useful when we want to use a constant decay value throughout the training